src/views/pages/Diagnostics/index.js

```diff
@@ -0,0 +1,55 @@
+import React, { useState, useEffect } from "react";
+import Node from '../../../services/Node';
+
+function Diagnostics() {
+
+    const [data, setData] = useState({ pending: true, diagnostics: null });
+
+    useEffect(() => {
+        setData({ pending: true, diagnostics: {} });
+
+        const fetchData = async () => {
+            const node = new Node();
+            const data = await node.getDiagnosticsData();
+            return data;
+        }
+        fetchData().then((fetchedData) => {
```

Comment:
You could use BehaviorSubject within the node class and subscribe to the observable here -- that way, any other component that references the Node service will share that same data

Reply:
My only question here is whether treating this as undefined is cleaner in the long run. I tend to prefer explicit over implicit and leaving it undefined might pose a risk of raising questions or wonky behavior later. Would doing it this way be significantly cleaner do handle in a component than explicitly setting the state as pending?

Reply:
Did you mean to reply to my other comment?

Reply:
I'd say that "undefined" is explicit. Either it will have data or it won't. Adding the pending variable doesn't guarantee that "data" has a value. It's like killing two birds with one stone.

Reply:
And yes, I prefer setting initial state to something like undefined rather than adding a "pending" variable around the actual data I care about because I think it's cleaner.

Reply:
I did mean to respond to the other one. Sorry about that. And sure, makes sense. I'm fine with that as long as it's done in a way that's concise and explicit which we're doing well on. We should add Typescript before too long, though. Working with undefined values in JS makes me uneasy. 😂